src/dataflow/src/storage/render/debezium.rs

```diff
@@ -118,6 +121,304 @@ where
+                                if event_count == 0 {
+                                    continue;
+                                }
+
+                                match tx_mapping.insert(tx_id.clone(), time) {
+                                    None => {
+                                        tx_event_count.insert(tx_id.clone(), event_count);
+                                        tx_cap_map
+                                            .insert(tx_id.clone(), tx_metadata_cap.clone());
+                                    }
+                                    Some(val) if val == time => {},
+                                    Some(val) => panic!("unexpected mismatch in duplicate END record for {:?}: {:?} vs {:?}", tx_id, time, val),
+                                }
+                            }
+                        }
+
+                        while let Some((data_cap, refmut_data)) = data_input.next() {
+                            // RefOrMut doesn't let us drain directly into iterator
+                            refmut_data.swap(&mut data);
+                            let data_cap = data_cap.retain();
+                            data_buffer.extend(data.drain(..).map(|r| (r, data_cap.clone())));
+                        }
+                        while let Some((result, data_cap)) = data_buffer.get(0).cloned() {
+                            let key = match result.key.transpose() {
+                                Ok(key) => key,
+                                Err(err) => {
+                                    output.session(&data_cap).give((
+                                        Err(err.into()),
+                                        *data_cap.time(),
+                                        1,
+                                    ));
+                                    let _ = data_buffer.pop_front();
+                                    continue;
+                                }
+                            };
+                            let value = match result.value {
+                                Some(Ok(value)) => value,
+                                Some(Err(err)) => {
+                                    output.session(&data_cap).give((
+                                        Err(err.into()),
+                                        *data_cap.time(),
+                                        1,
+                                    ));
+                                    let _ = data_buffer.pop_front();
+                                    continue;
+                                }
+                                None => {
+                                    let _ = data_buffer.pop_front();
+                                    continue;
+                                }
+                            };
+
+                            let tx_id = match value.iter().nth(transaction_idx).unwrap() {
```

Comment:
This variable is named `tx_id` but it actually contains both `tx_id` and `tx_time`